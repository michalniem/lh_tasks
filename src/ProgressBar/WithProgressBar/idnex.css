.progressContainer {
  max-height: 50vh;
  overflow: auto;
  position: relative;
}

.progressBar {
  position: sticky;
  height: 5px;
  top: 0;
  background: #ebebeb;
}

.progressBar span {
  height: 100%;
  display: block;
  background: blue;
}
