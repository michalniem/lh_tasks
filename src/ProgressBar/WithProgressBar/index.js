import React from "react";

import "./idnex.css";

function WithProgressBar(Component) {
  return class extends React.Component {
    state = {
      progress: 0,
    };

    handleScroll = (event) => {
      const { target: { scrollTop, scrollHeight, clientHeight }} = event;
      const componentHieght = scrollHeight - clientHeight;
      const percentageScrollProgress = (scrollTop / componentHieght) * 100;

      this.setState({ progress: percentageScrollProgress });
    };

    render() {
      return (
        <div className="progressContainer" onScroll={this.handleScroll}>
          <div className="progressBar">
            <span style={{ width: `${this.state.progress}%` }} data-test-id="progress_bar" />
          </div>
          <Component />
        </div>
      );
    }
  };
}

export default WithProgressBar;
