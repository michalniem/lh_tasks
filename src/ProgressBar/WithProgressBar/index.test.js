import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import WithProgressBar from "./index";

const NoopComponent = () => <div />;

describe("WithProgressBar", () => {
  const Component = WithProgressBar(NoopComponent);
  const wrapper = shallow(<Component />);
  const instance = wrapper.instance();

  beforeEach(() => {
    const event = {
      target: {
        scrollTop: 1200,
        scrollHeight: 2800,
        clientHeight: 400,
      },
    };
    instance.handleScroll(event);
  });

  test("should change progress state with proper value", () => {
    expect(wrapper.state("progress")).toEqual(50);
  });

  test("should set width style based on progress state", () => {
    const progressBarStyle = wrapper
      .find("[data-test-id='progressBar']")
      .prop("style");
    expect(progressBarStyle).toHaveProperty("width", "50%");
  });

  test("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
