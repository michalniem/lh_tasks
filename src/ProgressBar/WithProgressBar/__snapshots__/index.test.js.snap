// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithProgressBar should render correctly 1`] = `
<div
  className="progress__container"
  onScroll={[Function]}
>
  <div
    className="progress__bar"
  >
    <span
      data-test-id="progress_bar"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <NoopComponent />
</div>
`;
